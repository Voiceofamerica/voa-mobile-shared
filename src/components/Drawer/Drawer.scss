
@import "../../commonStyles/index";

.drawer {
  display: flex;
  position: fixed;
  flex-direction: column;
  top: $menu-height;
  left: 0;
  right: 0;
  bottom: 0;
  overflow: hidden;
  transform: translateY(100vh);
  transition: transform 0.3s;
  background: white;
  border-radius: 10px 10px 0 0;
  z-index: $menu-z-max;

  .drawerOpen & {
    transform: translateY(0);
  }

  .closeDrawer {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    text-align: center;
    font-size: 30px;
    height: 40px;
    color: #888;
    background: white;
    z-index: $menu-z + 5;

    i::before {
      transform: scale(2, 1)
    }
  }

  .content {
    overflow-y: scroll;
    -webkit-overflow-scrolling: touch;
    padding: 40px 0;
    display: flex;
  }
}

.overlay {
  background: transparent;
  z-index: $menu-z-max;
  transition: 0.3s background;

  .drawerOpen & {
    position: fixed;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    background: rgba(0, 0, 0, 0.3);
  }
}