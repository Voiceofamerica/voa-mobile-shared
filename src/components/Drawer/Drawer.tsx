
import * as React from 'react'

import {
  drawer,
  drawerOpen,
  content,
  closeDrawer,
  overlay,
} from './Drawer.scss'

export interface Props {
  open: boolean
  onClose: () => void
  className?: string
  style?: React.CSSProperties
}

class Drawer extends React.Component<Props> {

  render () {
    const { open, onClose, className = '', style = {}, children } = this.props
    const fullClassName = open ? `${drawer} ${drawerOpen}` : drawer

    return (
      <div className={open ? drawerOpen : ''}>
        <div className={overlay} onClick={() => onClose()} />
        <div className={`${drawer} ${className}`}>
          <div className={closeDrawer} onClick={() => onClose()}>
            <i className='mdi mdi-chevron-down' />
          </div>
          <div className={content}>
            {children}
          </div>
        </div>
      </div>
    )
  }
}

export default Drawer
