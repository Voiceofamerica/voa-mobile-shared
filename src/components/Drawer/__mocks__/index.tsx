
import * as React from 'react'

export default (props) => (
  React.createElement('drawer', { props }, props.children)
)
