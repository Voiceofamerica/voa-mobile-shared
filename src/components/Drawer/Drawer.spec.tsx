
import * as React from 'react'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'

import Drawer from './Drawer'
import { closeDrawer, overlay } from './Drawer.scss'

jest.mock('../ResilientImage')
jest.mock('../Spinner')

describe('<Drawer />', () => {
  describe('shapshots', () => {
    it('should render closed', () => {
      const element = create((
        <Drawer open={false} onClose={jest.fn()}>
          This should show up
        </Drawer>
      ))

      expect(element).toMatchSnapshot()
    })
    it('should render open', () => {
      const element = create((
        <Drawer open={true} onClose={jest.fn()}>
          This should show up
        </Drawer>
      ))

      expect(element).toMatchSnapshot()
    })
  })

  describe('onClose', () => {
    it('should be called whenever the close element is clicked', () => {
      const onClose = jest.fn()

      const wrapper = shallow((
        <Drawer open={true} onClose={onClose}>
          This should show up
        </Drawer>
      ))

      wrapper.find(`.${closeDrawer}`).simulate('click')

      expect(onClose).toHaveBeenCalledTimes(1)
    })
    it('should be called whenever the overlay element is clicked', () => {
      const onClose = jest.fn()

      const wrapper = shallow((
        <Drawer open={true} onClose={onClose}>
          This should show up
        </Drawer>
      ))

      wrapper.find(`.${overlay}`).simulate('click')

      expect(onClose).toHaveBeenCalledTimes(1)
    })
  })
})
