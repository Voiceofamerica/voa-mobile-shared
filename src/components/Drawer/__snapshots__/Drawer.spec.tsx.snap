// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> shapshots should render closed 1`] = `
<div
  className=""
>
  <div
    className="overlay"
    onClick={[Function]}
  />
  <div
    className="drawer "
  >
    <div
      className="closeDrawer"
      onClick={[Function]}
    >
      <i
        className="mdi mdi-chevron-down"
      />
    </div>
    <div
      className="content"
    >
      This should show up
    </div>
  </div>
</div>
`;

exports[`<Drawer /> shapshots should render open 1`] = `
<div
  className="drawerOpen"
>
  <div
    className="overlay"
    onClick={[Function]}
  />
  <div
    className="drawer "
  >
    <div
      className="closeDrawer"
      onClick={[Function]}
    >
      <i
        className="mdi mdi-chevron-down"
      />
    </div>
    <div
      className="content"
    >
      This should show up
    </div>
  </div>
</div>
`;
